### PR TITLE
Allow .jscsrc extensions for "preset" value and improve documentation for it

### PR DESCRIPTION
**Pull request by markelog**

### CODE REVIEW

File: OVERVIEW.md [hunk 1]
```diff
@@ -221,12 +221,19 @@ Extends defined rules with preset rules.
 
 Type: `String`
 
-Values: `"airbnb"`, `"crockford"`, `"google"`, `"jquery"`, `"mdcs"`, `"node-style-guide"`, `"wikimedia"`, `"wordpress"`, `"yandex"`
+Values: You can choose one of the default presets: `"airbnb"`, `"crockford"`, `"google"`, `"jquery"`, `"mdcs"`, `"node-style-guide"`, `"wikimedia"`, `"wordpress"`, `"yandex"`.
+
+Or you can load from it local path or as node module
```
Comment (mrjoelkemp) [inline OVERVIEW.md]: `Or you can load it from a local path or node module:`

File: OVERVIEW.md [hunk 2]
```diff
@@ -221,12 +221,19 @@ Extends defined rules with preset rules.
...
 
 #### Example
 
 ```js
 "preset": "jquery"
+"preset": "./path-to-your-preset"
+
+// If your preset called "jscs-your-preset-node_modules-path" 
+// You can either define full name or omit "jscs-" prefix -
```
Comment (mrjoelkemp) [inline OVERVIEW.md]: I don't really see the point to the sugar of `jscs-`. It only helps users who want to fork the built-in presets and retain the name. But if they're forking it, they can name it whatever they'd like and just supply that as the preset value.

Any big wins with this prefix?

File: lib/cli-config.js [hunk 3]
```diff
@@ -72,7 +76,15 @@ exports.getContent = function(config, directory) {
                 )
             );
         }
+    } else {
+
+        // Try to load it as a node module
+        try {
+            content = require(config);
+        } catch (e) {}
```
Comment (mrjoelkemp) [inline lib/cli-config.js]: What will the user see if the resolution fails? Said differently, what does the user see if this function returns `undefined`? I assume it'll bubble to cli.js and then show an error about a missing config?

File: test/specs/config/node-configuration.js [hunk 4]
```diff
@@ -127,6 +127,26 @@ describe('modules/config/node-configuration', function() {
...
 
+        it('should load external preset with .jscsrc extension', function() {
+            configuration.registerDefaultRules();
+
+            configuration.load({
+                preset: path.resolve(__dirname + '/../../data/configs/jscsrc/external.jscsrc')
+            });
+
+            var exist = false;
+            configuration.getRegisteredRules().forEach(function(rule) {
```
Comment (mrjoelkemp) [inline test/specs/config/node-configuration.js]: Use `.some` instead

**Comment by mrjoelkemp** [comment]
Test fail, but LGTM otherwise

File: OVERVIEW.md (hunk 2 above)
Comment (markelog) [inline OVERVIEW.md]: Prefixes usually added so they wouldn't be confused with other packages since it name without prefix might be misleading and have possibility of the name conflict - https://twitter.com/ironshay/status/370525864523743232

ESlint recent change of the same nature and popularity of https://github.com/sindresorhus/load-grunt-tasks packages somewhat proves it, plus it seems popular for the jscs too - https://www.npmjs.com/search?q=jscs

So we know ppl use those prefixes a lot, so we might take advantage of it...

And you know, it is nicer to write `preset: my-company`, then `preset: jscs-my-company`.

File: lib/cli-config.js (hunk 3 above)
Comment (markelog) [inline lib/cli-config.js]: Not exactly, but yeah

File: test/specs/config/node-configuration.js (hunk 4 above)
Comment (markelog) [inline test/specs/config/node-configuration.js]: Sure, although it seems logical to do in the different commit, since we use `forEach` in the test above

**Comment by markelog** [comment]
Hm, it successes with `mocha` command but fails with `npm run coverage`.

@mdevils could you help us out?

**Comment by mdevils** [comment]
Looks like a bug in unit-coverage, will fix it

File: OVERVIEW.md (hunk 2 above)
Comment (mrjoelkemp) [inline OVERVIEW.md]: Fair points. +1
